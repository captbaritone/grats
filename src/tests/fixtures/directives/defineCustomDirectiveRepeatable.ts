/**
 * This is my custom directive.
 * @gqlDirective repeatable on FIELD_DEFINITION
 */
function customDirective() {}
