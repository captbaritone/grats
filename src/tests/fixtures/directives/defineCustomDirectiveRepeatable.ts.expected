-----------------
INPUT
----------------- 
/**
 * This is my custom directive.
 * @gqlDirective repeatable on FIELD_DEFINITION
 */
function customDirective() {}

-----------------
OUTPUT
-----------------
-- SDL --
"""This is my custom directive."""
directive @customDirective repeatable on FIELD_DEFINITION
-- TypeScript --
import { GraphQLSchema, GraphQLDirective, DirectiveLocation } from "graphql";
export function getSchema(): GraphQLSchema {
    return new GraphQLSchema({
        directives: [new GraphQLDirective({
                name: "customDirective",
                locations: [DirectiveLocation.FIELD_DEFINITION],
                description: "This is my custom directive.",
                isRepeatable: true
            })],
        types: []
    });
}
